frontend/src/Student/components/ShowCard.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react'
+import styled from 'styled-components'
+import { Link } from 'react-router-dom'
+import PropTypes from 'prop-types'
+import Moment from 'react-moment'
+import FaPlusCircle from 'babel-loader!react-icons/fa/plus-circle'
+import { Button, ButtonGroup, Row, Col } from 'reactstrap'
+
+const Card = styled.div`
+  background-color: #f8f9fa;
+  border-radius: 5px;
+  margin-top: 15px;
+  padding: 10px;
+  width: 100%;
+`
+
+const ShowCard = (props) => (
+  <Card>
+    <Row>
+      <Col>
+        <h2>{props.name}</h2>
+      </Col>
+      <Col className="text-right">
+      <div>
+        <h5>{props.entries ? props.entries.length : 0}/{props.entryCap} submissions</h5>
+      </div>
+      <div>
+        Submission Ends: <Moment format='YYYY/MM/DD'>{props.entryEnd}</Moment>
+      </div>
+      </Col>
+    </Row>
+    <hr />
+    <Row>
+      <Col className='text-center align-self-center h-100'>
+       <Link to={`/submit/${props.id}`} className='d-block w-100 h-100'>
+         <FaPlusCircle size='3em' />
+         <h5>New Submission</h5>
+       </Link>
+      </Col>
+    </Row>
+  </Card>
+)
+
+ShowCard.propTypes = {
```

Comment:
Instead of having all these props passed in shallowly, I would encapsulate them in a `show` prop:

```js
ShowCard.propTypes = {
  show: PropTypes.shape({
    id: PropTypes.string,
    ...
    judgingEnd: PropTypes.string
  }).isRequired
}
```

See comment about `entries` below